devicetypes/smartthings/zigbee-ceiling-fan-light.src/itm-fan-child.groovy

```diff
@@ -0,0 +1,97 @@
+/*
+ *  Copyright 2017 SmartThings
+ *
+ *  Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except
+ *  in compliance with the License. You may obtain a copy of the License at:
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *  Unless required by applicable law or agreed to in writing, software distributed under the License is distributed
+ *  on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the License
+ *  for the specific language governing permissions and limitations under the License.
+ *
+ *  ZigBee Ceiling Fan Light
+ *
+ *  Author: SAMSUNG LED
+ *  Date: 2021-06-30
+ */
+
+import physicalgraph.zigbee.zcl.DataType
+import groovy.json.JsonOutput
+
+metadata {
+	definition(name: "ITM Fan Child", namespace: "SAMSUNG LED", author: "SAMSUNG LED", ocfDeviceType: "oic.d.fan") {
+		capability "Actuator"
+        	capability "Configuration"
+        	capability "Refresh"
+        	capability "Switch"
+		/* Capability "Switch Level" is used to control fan speed for platforms don't support capability "Fan speed"
+		 * when you connect other platforms via SmartThings cloud to cloud connection. */
+		capability "Switch Level"
+        	capability "Fan Speed"
+    }    
+}
+
+def off() {
+	setFanSpeed(0)
+}
+
+def on() {
+	setFanSpeed(1)
+}
+
+def setLevel(value){
+	if (value <= 1) {
+		setFanSpeed(0)
+        } else if (value <= 25) {
+		setFanSpeed(1)
+        } else if (value <= 50) {
+		setFanSpeed(2)
+        } else if (value <= 75) {
+		setFanSpeed(3)
+	} else if (value <= 100) {
+		setFanSpeed(4)
+       	}	
+}
+
+def setFanSpeed(speed) {
+	log.debug "child setFanSpeed $speed"	
+	if (speed as Integer == 0) {
+        	sendEvent(name: "switch", value: "off", displayed: true, isStateChange: true)
+    	} else {
+        	sendEvent(name: "switch", value: "on", displayed: true, isStateChange: true)
+    	}	
+	parent.setFanSpeed(speed, device)
```

Comment:
@RaihaPark have you thought about this?

Reply:
@RaihaPark I'd really prefer if you'd address it. At least explain why do you think it's not a good idea to change it.

Reply:
Oh, I'm sorry. I missed it. But we designed this code parents side have all control functions. So there is a parse function only in the parents side. Because of other platform issue we have to did that, especially alexa. If we call directly fanspeed function in the child side as like line 146, the fan controls well but we see some error message together. So we had to separate control functions to parent side.

Reply:
I think we don't understand each other. 
Let's assume there's a UC - user sets fan speed to 75% through Alexa.
Current flow is:
1. Alexa calls, with SmartThings API, child device's `Switch Level`'s command `setLevel(value)` with `value = 75`.
2. Child DTH calls `setFanSpeed(speed)` with `speed = 3`. 
3. We're still in Child DTH - now parent's method is called - `parent.setFanSpeed(speed, device)`
4. We're in parent DTH, in `setFanSpeed(value, device)` method, where `value = 3` and `sendFanSpeed(val)` is called.
5. In `sendFanSpeed(val)`, finally, a zigbee command is send.

If you've done what I want you to do, it'll be reduced to:
1. Alexa calls, with SmartThings API, child device's `Switch Level`'s command `setLevel(value)` with `value = 75`.
2. Child DTH calls `setFanSpeed(speed)` with `speed = 3`. 
3. Parent's method is called - `parent.sendFanSpeed(speed)`
4. We're in parent DTH, in `setFanSpeed(value, device)` method, where `value = 3` and `sendFanSpeed(val)` is called.

You remove one extra step, but what's more important you clean up the code big time. 
In parent DTH you really don't need `setFanSpeed()`, it's only called from child DTH as parent doesn't have `Fan Speed` now.
All you need to do is remove `setFanSpeed(speed, device)` from parent DTH and change 64 line in child DTH from
```
parent.setFanSpeed(speed, device)
```
to
```
parent.sendFanSpeed(speed)
```
and you'd have same functionality with less pain.